doc/source/about/roadmap.md

```diff
@@ -1,5 +1,93 @@
 # Roadmap
 
+This page provides an overview of the strategic goals for development of GeoPandas.
+Some of the tasks may happen sooner given the appropriate funding,
+other later with no specified date, and some may not happen at all if the implementation
+proves to be against the will of the community or face technical issues preventing their
+inclusion in the code base.
+
+The current roadmap is divided into two milestones. The first milestone aims at a
+release of the first major version of GeoPandas, while the second milestone is a longer-term
+vision covering enhancements that should happen in subsequent releases.
+
 ## Roadmap for GeoPandas 1.0
 
-WIP
+### Fully vectorized geometry engine
+
+GeoPandas uses `shapely` as its geometry engine, based on scalar geometries, requiring a
+loop-based implementation of most GeoPandas methods. That comes at a significant
+performance cost, which is being resolved in shapely 2.0, a new major release resulting
+from a complete rewrite of the internals using the vectorized implementation prototyped
+in the `PyGEOS` project. At this moment, GeoPandas supports `shapely<2.0`,
+`shapely>=2.0`, and `PyGEOS` as possible geometry engines, which causes friction in the
+development process and uneven performance on the user side based on what geometry engine the user happens to be using.
+
+GeoPandas 1.0 will require `shapely>=2.0` and deprecate both older shapely and `PyGEOS`
+engines. This change should simplify the code base allowing more manageable maintenance
+and a lower barrier to entry for new contributors.
+
+### Feature parity with shapely
+
+Even though GeoPandas uses shapely as the geometry engine, not all its functions are
+exposed at a GeoPandas level. This has resulted in a less convenient API and a need to switch
+between `GeoSeries` objects and lists or arrays of geometries, potentially risking the
+data loss or corruption as the CRS is not included in such operations. In the first
+phase, all element-wise operations (e.g. `segmentize`, or `minimum_bounding_circle`)
+should be exposed as `GeoSeries` methods. The feature parity should be reached in the
+second phase, covering all relevant functions.
+
+### Clarity of the API
+
+The first version of the GeoPandas API is nearly ten years old. The PyData ecosystem has
+significantly changed in the meantime, and some of the early decisions may no longer be future-proof.
+Ahead of GeoPandas 1.0, the API will be revised to ensure
+that all the necessary deprecations occur before the major release to provide the
+stability of the API for the coming years.
+
+### Prepared geometries
```

Comment:
We _certainly_ should look into this, but maybe this is more something for beyond 1.0?

Reply:
Yeah, I was not sure where to put this one. I remember @brendan-ward suggesting it for 1.0 but we may potentially leave it for later. I'll move it below.